lib/cereal/workflow_scheduler.go

```diff
@@ -86,36 +82,45 @@ func (w *workflowScheduler) scheduleWorkflow(ctx context.Context) (time.Duration
 		logrus.Warnf("Recurring workflow %fs past due. (expected at %s)", time.Since(s.NextDueAt).Seconds(), s.NextDueAt)
 	}
 
-	// TODO(ssd) 2019-05-13: We might need two different
-	// rule types here to support the different use cases.
 	recurrence, err := rrule.StrToRRule(s.Recurrence)
 	if err != nil {
-		// TODO(ssd) 2019-05-13: Perhaps we should disable this rule so that it doesn't keep producing errors
-		// We need to do this otherwise we'll just keep trying to enqueue this and nothing else
-		return maxWakeupInterval, errors.Wrap(err, "could not parse recurrence rule for workflow, skipping")
+		logrus.WithError(err).Error("scheduled workflow has invalid recurrence rule, attempting to disable it")
+		err2 := completer.DisableSchedule(s)
+		if err2 != nil {
+			return maxWakeupInterval, errors.Wrap(err, "failed to disable workflow with invalid recurrence rule")
```

Comment:
err2?

Reply:
I've gone back and forth, hence the confusion at this site at the moment.  Since we are logging the first error at 87, let's return err2 here but not log it since the returned error will result in a log line.